packages/core/src/httpClient/basicHttpClient.ts

```diff
@@ -215,7 +216,7 @@ export class BasicHttpClient {
   protected async request<ResponseType>(request: HttpRequest) {
     const mutatedRequest = await this.preRequest(request);
     const rawResponse = await this.rawRequest<ResponseType>(mutatedRequest);
-    return this.postRequest(rawResponse, mutatedRequest);
+    return this.postRequest(rawResponse, request, mutatedRequest);
```

Comment:
We need a way to differentiate between default headers and original headers in the request. Filtering out the default headers will not work because they can be overridden in the original request.

We need the mutated request to compare the authorization header used in the request, so maybe we can pass that instead of whole request.